src/services/api/LocalStorageService.ts

```diff
@@ -0,0 +1,24 @@
+import { IClause } from "configurations/apiSchema";
+
+const DEALS_DATA = "DEALS_STORAGE_V1";
+const CLAUSES_DATA = "DEALS_STORAGE_V1";
+
+export class LocalStorageService {
+  public getClausesData(): IClause[] {
+    const clausesData = this.getItem(CLAUSES_DATA);
+    return clausesData as unknown as IClause[];
+  }
+
+  private getItem(key = DEALS_DATA): Record<string, unknown> {
```

Comment:
> We can use typescript generics to enforce the return type of this method.

Yeah, would make sense over `as unknown as XYZ` for the usage place.

> memoization

Feel free. But honestly, we are communicating with the local storage here, and this might be over-optimization.

Looking at the whole Deals app, I don't see the immediate need for memoization either.
Anyway, open to discuss this on PRs (continue here, or in others)

Reply:
> Yeah, would make sense over as unknown as XYZ for the usage place.

Feel free to mark this as resolved when ready.


> memoization

got it. let's leave it as it is